src/HumusAmqpModule/Consumer.php

```diff
@@ -263,42 +263,40 @@ public function consume($msgAmount = 0)
     {
         $this->target = $msgAmount;
 
-        foreach ($this->queues as $index => $queue) {
-            if (!$this->timestampLastAck) {
-                $this->timestampLastAck = microtime(1);
-            }
-
-            $message = $queue->get();
+        /** @var \AMQPQueue $queue */
+        foreach($this->queues as $queue) {
+            $queue->consume(function ($message, $queue) {
```

Comment:
I'm still can't imagine usage with several queues per one consumer.
Anyway, what if I will add 'ConsumerType' parameter to config with values "get" or "consume" and depending on this parameter factory will create proper consumer. By default will be "get" of course.